core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,6 +142,45 @@ oppia.factory('SimpleEditorManagerService', [
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      changeQuestion: function(type, index) {
+        var currentStateName = (
+          index === 0 ? 'Introduction' : 'Question ' + index);
+        var questionCount = data.questionList._questions.length;
+        var answerGroupsLength = (
+          data.questionList._questions[index]._answerGroups.length);
+        var nextStateName = 'Question ' + (index + 1);
+        var currentInteractionId = (
+          data.questionList._questions[index]._interactionId);
+
+        // Update Question Type If interactionId is not same.
+        if(type !== currentInteractionId) {
+          var newAnswerGroups = [];
+
+          switch (type) {
+            case 'MultipleChoiceInput':
+              SimpleEditorShimService.saveInteractionId(
+                currentStateName, DEFAULT_INTERACTION.ID);
+              SimpleEditorShimService.saveCustomizationArgs(
+                currentStateName, DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+              newAnswerGroups.push(AnswerGroupObjectFactory.createNew([
+                RuleObjectFactory.createNew('Equals', {
+                  x: 0
+                })
+              ], OutcomeObjectFactory.createEmpty(nextStateName), false));
+              break;
+          }
+          if(answerGroupsLength !== 0 && index !== questionCount - 1) {
+            SimpleEditorShimService.saveAnswerGroups(currentStateName
+              , newAnswerGroups);
+          }
+        }
+        // Update the Question Set.
+        var questions = StatesToQuestionsService.getQuestions();
+        if (!questions) {
+          return false;
+        }
+        data.questionList = QuestionListObjectFactory.create(questions);
```

Comment:
Why do we need to create the questionList from scratch, rather than append a new question (or modify an existing one)?

Also, does this handle the question whose type is being changed is somewhere in the middle of the sequence?